src/proxy.js

```diff
@@ -262,6 +272,91 @@ function iterateMapValues(state, prop, receiver) {
     }
 }
 
+/**
+ * Set drafts
+ */
+
+const setGetters = {
+    [DRAFT_STATE]: state => state,
+    size: state => source(state).size,
+    has(state, prop) {
+        state = source(state)
+        return state.has.bind(state)
+    },
+    add: state => value => {
+        if (!source(state).has(value)) {
+            markChanged(state)
+            state.copy.add(value)
+        }
+        return state.draft
+    },
+    delete: state => value => {
+        markChanged(state)
+        return state.copy.delete(value)
+    },
+    clear: state => () => {
+        markChanged(state)
+        return state.copy.clear()
```

Comment:
Does this need to mutate `state.assigned`?

Reply:
For now it seems like we don't need `assigned` for ES6 Sets at all. It's used for other data structures to generate 'replace' patches, but we cannot really replace anything in a Set, we can only remove and add.